woocommerce/woocommerce
# FlexSlider uses Web Animations API for "slide" instead of $.animate()

The basic Web Animations API functionality is ready for use in WooCommerce-supportred browsers:

- **Element.animate():** https://caniuse.com/mdn-api_element_animate
- **Animation finish event:** https://caniuse.com/mdn-api_animation_finish_event

This PR only affects the **"slide"** animation for FlexSlider (which is the WooCommerce default); the **"fade"** animation is unaffected.

This PR does NOT come into effect with the default WooCommerce config, which is the **"slide"** animation type + `(!video && useCSS)` - the PR only changes the behavior when either `video` or `!useCSS` is true.

**if `!video` and `useCSS` (WooCommerce defaults):**

- Before this PR:
  - `"translate3d"` is updated with CSS3 `transform` transitions
  - no easing method
- After this PR:
  - `"translate3d"` is updated with CSS3 `transform` transitions
  - no easing method

**if `video` or `!useCSS`:**
- Before this PR:
  - `top`/`marginLeft`/`marginRight` (one of those is stored in `slider.prop`) is animated with `$.animate()`
  - jQuery easing
- After this PR:
  - `top`/`marginLeft`/`marginRight` (one of those is stored in `slider.prop`) is animated with `Element.animate()`
  - we mimic jQuery easing methods via the `easing` option (see https://easings.net/)

Technically, this is a major change, as this PR removes support for the following funky easings from the [jQuery easing plugin](https://gsgd.co.uk/sandbox/jquery/easing/):

1. easeInElastic (https://easings.net/#easeInElastic)
2. easeOutElastic (https://easings.net/#easeOutElastic)
3. easeInOutElastic (https://easings.net/#easeInOutElastic)
4. easeInBounce (https://easings.net/#easeInBounce)
5. easeOutBounce (https://easings.net/#easeOutBounce)
6. easeInOutBounce (https://easings.net/#easeInOutBounce)

The remaining 24 easings from the [jQuery easing plugin](https://gsgd.co.uk/sandbox/jquery/easing/) are still supported.

In addition, we no longer need to load the [jQuery easing plugin](https://gsgd.co.uk/sandbox/jquery/easing/) for the **"slide"** animation type (the WooCommerce default).

This PR has no effect on the **"fade"** animation type - it will still use `$.animate()` on non-touch-enabled devices.

### All Submissions:






### Changes proposed in this Pull Request:



Fixes #36994.



### How to test the changes in this Pull Request:



1. Add this custom code to your theme's `functions.php`:
```
function carousel_options( $options ) {
	return array_merge(
		$options,
		array(
			'animationSpeed' => 1600,
			'easing' => 'easeInOutBack',
			'video' => true
		)
	);
}
add_filter( 'woocommerce_single_product_carousel_options', 'carousel_options' );
```

2. Go to a Single Product Page.
3. Click on different carousel image thumbnails.
4. Confirm that animation for switching between images takes about 1.6s and transition looks like in the GIF below:

![cssflex](https://user-images.githubusercontent.com/3998720/221844554-d854c5db-cca9-4e09-a7fc-f455e42c68f0.gif)if



### FOR PR REVIEWER ONLY:
